### DISCUSSION TITLE
Freerdp3 from master branch or freerdp2 release. Which one should I choose?

### DISCUSSION THREAD

**Discussion by astepanov83** [FreeRDP/FreeRDP#8784] [Q&A] [answered]
Hi.

I'm developing an RDP proxy. I'm using freerdp2 from release tags, and when you roll out a new version, I checkout it from a tag, then build everything, test my app, and release. All was fine until a couple of months ago, when I started implementing drives proxying. I realized that there was no code ready to use like, say, for clipboard proxying, but I found it in freerdp3. So I ended up copying code from freerdp3 master branch, and also patching freerdp2 core files here and there. It was extremely ugly, but it worked, and I was happy with this solution.

Still, I was thinking whether I should port my code to freerdp3, or stay with freerdp2. On the one hand, freerdp2 releases are deemed to be more stable, unlike code taken from master branch, which is constantly evolving. On the other hand, freerdp3 looks to be more feature-rich. Like I noticed, that RDP Correlation Info is supported in v3, but is absent in v2. Although I don't use it, I'm not sure that there are more things like this, and like drives proxying, which I may need in future.

So what should I prefer? Stable code is in favor, although I've already mixed in pieces of "master" code to the release code in my project. I would appreciate any suggestions and comments.

Thanks,
Alex.

**Comment by akallabeth** [comment, marked as answer]
@astepanov83 while `stable-2.0` is stable, it does not get feature updates (only essential fixes required to be useful)
we´re currently working on getting a `3.0` prerelease out in the next couple of months (depending on time available and regressions we have to fix) so it might be a good idea to start supporting our development version.

(note: most of the `API` breaking stuff should have been done, but I can´t guarantee that until we´ve released 3.0 )

**Reply by astepanov83** [reply]
@akallabeth sounds great
ok, I think it's worth to start shifting to freerdp3
thanks
